src/hammer-vlsi/cli_driver_test.py

```diff
@@ -39,6 +42,199 @@ def generate_dummy_config(self, syn_rundir: str, config_path: str, top_module: s
+        top_module = "dummy"
+        config_path = os.path.join(syn_rundir, "run_config.json")
+        syn_out_path = os.path.join(syn_rundir, "syn_out.json")
+        syn_to_par_out_path = os.path.join(syn_rundir, "syn_par_out.json")
+        self.generate_dummy_config(syn_rundir, config_path, top_module)
+
+        self.run_syn_to_par_with_output(config_path, syn_rundir, par_rundir,
+                                        syn_out_path, syn_to_par_out_path)
+
+        # synthesis output should NOT keep other settings
+        with open(syn_out_path, "r") as f:
+            syn_output = json.loads(f.read())
+            self.assertEqual(syn_output["synthesis.outputs.output_files"], [])
+            self.assertFalse("vlsi.core.technology" in syn_output)
+        # Generated par input should have other settings
+        with open(syn_to_par_out_path, "r") as f:
+            par_input = json.loads(f.read())
+            self.assertEqual(par_input["par.inputs.top_module"], top_module)
+            # par-input should preserve other settings
+            self.assertEqual(par_input["vlsi.core.technology"], "nop")
+
+        # Cleanup
+        shutil.rmtree(syn_rundir)
+        shutil.rmtree(par_rundir)
+
+    def test_syn_to_par_full(self) -> None:
+        """
+        Test that syn-to-par works with the full JSON.
+        """
+
+        # Set up some temporary folders for the unit test.
+        syn_rundir = tempfile.mkdtemp()
+        par_rundir = tempfile.mkdtemp()
+
+        # Generate a config for testing.
+        top_module = "dummy"
+        config_path = os.path.join(syn_rundir, "run_config.json")
+        syn_out_full_path = os.path.join(syn_rundir, "syn-output-full.json")
+        syn_to_par_out_path = os.path.join(syn_rundir, "syn_par_out.json")
+        self.generate_dummy_config(syn_rundir, config_path, top_module)
+
+        # Check that running the CLIDriver executes successfully (code 0).
+        with self.assertRaises(SystemExit) as cm:  # type: ignore
+            CLIDriver().main(args=[
+                "syn", # action
+                "-p", config_path,
+                "--syn_rundir", syn_rundir,
+                "--par_rundir", par_rundir
+            ])
+        self.assertEqual(cm.exception.code, 0)
+
+        # Now run syn-to-par with the main config as well as the outputs.
+        with self.assertRaises(SystemExit) as cm:  # type: ignore
+            CLIDriver().main(args=[
+                "syn-to-par", # action
+                "-p", syn_out_full_path,
+                "--output", syn_to_par_out_path,
+                "--syn_rundir", syn_rundir,
+                "--par_rundir", par_rundir
+            ])
+        self.assertEqual(cm.exception.code, 0)
+
+        # synthesis full output should keep other settings
+        with open(syn_out_full_path, "r") as f:
+            syn_output = json.loads(f.read())
+            self.assertEqual(syn_output["synthesis.outputs.output_files"], [])
```

Comment:
Again this seems reversed.

Reply:
See the line below ("other setting" like `vlsi.core.technology` is kept)